stencils/pace/stencils/testing/savepoint.py

```diff
@@ -0,0 +1,64 @@
+import dataclasses
+import os
+from typing import Dict, Protocol, Union
+
+import numpy as np
+import xarray as xr
+
+from .grid import Grid
+
+
+def dataset_to_dict(ds: xr.Dataset) -> Dict[str, np.ndarray]:
+    return {
+        name: _process_if_scalar(array.values) for name, array in ds.data_vars.items()
+    }
+
+
+def _process_if_scalar(value: np.ndarray) -> Union[np.ndarray, float, int]:
+    if len(value.shape) == 0:
+        return value.item()
+    else:
+        return value
+
+
+class Translate(Protocol):
```

Comment:
We actually use a lot more attributes but mypy doesn't complain. I can add the others if needed.

Reply:
Would it be better to not have this if we don't plan on adding all the attributes? I think it's more confusing if it only has a subset.

Reply:
I can change it to Any or add the attributes, either way.